docs/pages/guides/dom-components.mdx

```diff
@@ -154,6 +154,59 @@ getDeviceName(): Promise<string> {
 
 Think of these functions like React Server Actions, but instead of residing on the server, they live locally in the native app and communicate with the DOM component. This approach provides a powerful way to add truly native functionality to your DOM components.
 
+## Passing refs
+
+You can use the `useDomImperativeHandle` hook inside a DOM component to accept ref calls from the native side. This hook is similar to React's [`useImperativeHandle`](https://react.dev/reference/react/useImperativeHandle) hook, but it does not need a ref object to be passed to it.
+
+```tsx App.tsx (native)
+import { useRef } from 'react';
+import { Button, View } from 'react-native';
+
+import MyComponent, { type MyRef } from './my-componenet';
+
+export default function App() {
+  const ref = useRef<MyRef>(null);
+
+  return (
+    <View style={{ flex: 1 }}>
+      <MyComponent ref={ref} />
+      <Button
+        title="focus"
+        onPress={() => {
+          ref.current?.focus();
+        }}
+      />
+    </View>
+  );
+}
+```
+
+```tsx my-component.tsx (web)
+'use dom';
+
+import { useDomImperativeHandle, type DOMImperativeFactory } from 'expo/dom';
+import { useRef } from 'react';
+
+export interface MyRef extends DOMImperativeFactory {
+  focus: () => void;
+}
+
+export default function MyComponent(props: React.RefAttributes<MyRef>) {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useDomImperativeHandle(
```

Comment:
Does this API work on web? i.e. just `useImperativeHandle` on web?

Reply:
good point, will try to use `useImperativeHandle` as underlying implementation for web.